Feature/modules

maosan132 opened the pull request:
# Description

Hello pals, I am re-submitting this pull request so everybody can give support and suggestions for making this project better, so it can be approved. Any suggestions will be fully appreciated. 

# Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

maosan132 committed:
chore(TODO): :tada: Initial Files

Added required files such as settings.json, linters, readme, etc

maosan132 committed:
chore(TODO): :alien: Add webpack config file and sextensions for css importing

maosan132 committed:
chore(TODO): :hammer: Added webpack config file and css importing extension

maosan132 committed:
chore(TODO): :construction: Added files which corresponds to building blocks or modules

maosan132 committed:
Merge branch 'feature/initial-files' into develop

maosan132 committed:
feat(TODO): :boom: Started adding html templates for easy looping with data

maosan132 committed:
feat(TODO): :sparkles: Added code to make a simple task lists creator

maosan132 committed:
chore(TODO): :sparkles: Added html template for rendering projects

This is the place where the loop with data are gonna render to app

maosan132 committed:
chore(TODO): :sparkles: Added all html elements to index

The elements added are all ux elements

maosan132 committed:
chore(TODO): :sparkles: Working on html parts

added style and form to general views, every part inside index.html

maosan132 committed:
refactor(TODO): :sparkles: Added priority, date and description properties to task pbject

maosan132 committed:
refactor(TODO): :sparkles: Created next module files and refatored task creator

Added priority, date and description of myProject object

maosan132 committed:
style(TODO): :lipstick: Cleaned index.html and move html files to each module

maosan132 committed:
Merge branch 'feature/views' into develop

maosan132 committed:
feat(TODO): :sparkles: Added menu buttons event listener

maosan132 committed:
refactor(TODO): :zap: Added 3 modules, start working w/ classes

Moved to mvc to manage modules

maosan132 committed:
Deleted unused modules: project-renderer.js, project-creator.js, loader-modules.js

maosan132 committed:
chore(TODO): :art: Fixed ESLINT warnings and errors

voscarmv reviewed (changes requested):
Hi, I'm invalidating this project because:

- Your Stylelint linter tests are failing. Please fix this.
- Your README file does not have a proper screenshot of the project.

Please apply these changes and request a new code review.

Happy hacking!

maosan132 committed:
fix(TODO): crafted functions for render todo lists, creating the todo form, crating and worked on function createTasks

maosan132 committed:
working on function createTask and findPorject

maosan132 committed:
Resolved diffs between incoming and current changes in controller and model modules

maosan132 committed:
refactor(TODO): :sparkles: Split model mdule into task-model and project-model modules

In order to comply with separation of concerns, decided to work with a diff model for tasks and project

maosan132 committed:
refactor(TODO): :zap: Minor adjustments

Move a fom element assignment out of the renderer ( taskitems)

maosan132 committed:
refactor(TODO): :art: Created redndertasks function and other minor fixes

maosan132 committed:
refactor(TODO): :zap: Made object task inside taskList array has an extra key  for easy capture

maosan132 committed:
chore(TODO): :sparkles: Added functionality of status and delete actions for each task item

maosan132 committed:
refactor(TODO): :bug: Fixed two bugs in fx adDomElem and renderTodoItems

#addDomElem accept now empty strings and renderTodoItems effectively iterates on myProjects items to find tasks

maosan132 committed:
chore(TODO): :rocket: Added debug function to find a bug

maosan132 committed:
refactor(TODO): :bug: Make icons for tasks workable

Make changes in renderToDoItems so it cleans todo container every time icon is clicked

maosan132 committed:
fix(TODO): :bug: Fixed default project and explore projects

Improved functions so both options works as expected

maosan132 committed:
chore(TODO): :zap: Make minor changes

maosan132 committed:
Deleted unused files

maosan132 committed:
fix(TODO): :zap: Fixed a bug in renderTodoCOntainer

When trying to render the todo container for only one project, it showed mistakenly foraneous tasks.

maosan132 committed:
Fixed Done button, which wasnt working right

maosan132 committed:
refactor(TODO): :fire: Removed commented, unused and consolelogs code

maosan132 committed:
Fixed Eslint warnings and errors

maosan132 committed:
feat(TODO): :rocket: Add screenshot to readme

maosan132 committed:
Added screenshot to readme

maosan132 committed:
Updated readme file

maosan132 committed:
Trying to add localStorage, no luck: a little TSE help would be welcomed

maosan132 committed:
Eslint warning dealed

maosan132 committed:
style(TODO): :art: Moved css out of index to module

maosan132 committed:
some configuration job for stylelint which complains about not finding css files

maosan132 committed:
Fixed linters

trekab reviewed (commented):
#### Invalid! :stop_sign: 
Friendly reminder: You should use Git Flow in this project. Please take a step back and watch the video [here](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/59882084/). Then create a brand new repo with the correct setup and feature branch (with a DESCRIPTIVE name) and copy your code there. This Code Review Request will be marked as invalid, so please submit a new one once you are ready!

__Treasure Kabareebe__
[Portfolio](https://trekab.github.io/portfolio/) | [LinkedIn](https://www.linkedin.com/in/treasure-kabareebe/) | [Twitter](https://twitter.com/TKabareebe)
